docs/developer-docs/web-apps/verifiable-credentials/issuer.mdx

```diff
@@ -0,0 +1,184 @@
+
+An example implementation in Rust can be found below:
+
+```rust
+async fn derivation_origin(
+    req: DerivationOriginRequest,
+) -> Result<DerivationOriginData, DerivationOriginError> {
+    get_derivation_origin(&req.frontend_hostname)
+}
+
+fn get_derivation_origin(_hostname: &str) -> Result<DerivationOriginData, DerivationOriginError> {
+    CONFIG.with_borrow(|config| {
+        let config = config.get();
+        println!("*** derivation origin: {}", config.derivation_origin);
+        Ok(DerivationOriginData {
+            origin: config.derivation_origin.clone(),
+        })
+    })
+}
+```
+
+:::info
+The returned derivation origin is subject to verification via `.well-known/ii-alternative-origins`, as described in the [feature description](https://internetcomputer.org/docs/current/references/ii-spec#alternative-frontend-origins).
+:::
+
+### Step 3: Prepare the credentials (`prepare_credential`)
+
+Before a credential can be issued, the request for the credential must be checked for validity using the `prepare_credential` API endpoint. Once it has been validated, then the actual VC can be prepared. The request is validated through the `signed_id_alias.credential_jws` token. If this check is successful, the credential data is prepared and returned in `PreparedCredentialData.prepared_context`.
+
+An example implementation in Rust can be found below:
+
+```rust
+fn prepare_credential_jwt(
+    credential_spec: &CredentialSpec,
+    alias_tuple: &AliasTuple,
+) -> Result<String, IssueCredentialError> {
+    let (group_name, owner) = verify_spec_and_get_group_name(credential_spec)
+        .map_err(IssueCredentialError::UnsupportedCredentialSpec)?;
+    GROUPS.with_borrow(|groups| {
+        verify_principal_is_member(alias_tuple.id_dapp, group_name, owner, groups)
+    })?;
+    Ok(verified_member_credential(
+        alias_tuple.id_alias,
+        credential_spec,
+    ))
+}
+```
+
+:::info
+The value of `prepared_context` is used to transfer information between the `prepare_credential` and `get_credential` steps. It is up to the issuer API to decide on the content of this field since the issuer API creates `prepared_context` and is the only entity that consumes it.
+
+For example, when using [canister signatures](https://internetcomputer.org/docs/current/references/ic-interface-spec#canister-signatures)
+`prepared_context` contains a time stamped and unsigned VC, for which the canister signature will be available through the `get_credential` call.
+:::
+
+### Step 4: Issue the credential (`get_credential`)
+
+Lastly, the issuer API endpoint `get_credential` is used to issue the credential requested by the user. The same checks used to prepare the credential are ran again, with an additional verify step to confirm that `prepared_context` is consistent with `prepare_credential`. If successful, the issuer API returns the signed credential in JWS-format.
+
+An example implementation in Rust can be found below:
```

Comment:
Same as before, don't you think this is too long a code? And devs still need to understand the different parts and functions.

> An example implementation in Rust can be found in [the VC Playground code](https://github.com/dfinity/vc-playground/blob/b2f8cd5f5397b7b355b6503df5af3789834b79e9/issuer/src/main.rs#L603).

Reply:
Replaced code with link to playground code